### PR TITLE
C/C++ Standard Lib for contracts (part 1)

### PR DESCRIPTION
**Pull request by wanderingbort**
Not all of it works so far but the basic framing is there and all of our current contracts compile and tests run.

will continue to improve

### CODE REVIEW

File: contracts/eoslib/stlib.h [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#pragma once
+
+// forward declarations, to be filled in by the compiler, for placement new.
```
Comment (heifner) [inline contracts/eoslib/stlib.h]: This comment does not apply.

**Comment by heifner** [comment]
Build error on Travis:
```
ninja: error: '../contracts/eosiolib/eosiolib.bc', needed by 'contracts/eosio.system/eosio.system.bc', missing and no known rule to make it
```

**Comment by spoonincode** [comment]
Something seems wrong in the cmake rules. A make -j8 will fail the first couple of times but will finally make it through after a few failed builds.

**Comment by wanderingbort** [comment]
the new changes from @moskvanaft have resolved the build issues I've seen locally.   Waiting on travis and/or @spoonincode and/or @heifner  to confirm

**Comment by spoonincode** [comment]
Worked For Me

**Pull request merged**